src/server.rs

```diff
@@ -13,12 +12,18 @@ async fn ping() -> impl Responder {
 async fn send_email(event: web::Json<EmailMessage>) -> impl Responder {
     let event_data = event.into_inner();
     info!(LOG, "Received email to send: {:?}", event_data);
-    task::spawn(async move {
-        if let Err(e) = SMTP_CLIENT.get().unwrap().send_new_email(event_data).await {
+
```

Comment:
I'd like to know why did you remove task::spawn?

Reply:
Because I need to return the message_id in the API call. The message_id get's created in the send_new_email function. To wait for the message being sent and only after returning the message_id to the API caller with task:spawn would be a lot more complicated. Also tokio will spawn a new thread per request automatically too

Reply:
Also, by waiting for the send_new_email function to be successful, we can actually know if the message send was successful in the other services calling this API and handle that accordingly

Reply:
@DimiDumo Thank you for the clear answer. Understood.